### PR TITLE
Support Go Get Command - Update go tests

### PR DESCRIPTION
**Pull request by talarian1**
Added test for the new "go get" command.
fixed TestGoBuildInfo and TestGoPublishResolve.

- [x] All [tests](https://github.com/jfrog/jfrog-cli#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: go_test.go [hunk 1]
```diff
@@ -143,12 +145,39 @@ func TestGoWithGlobalConfig(t *testing.T) {
...
+	initGoTest(t)
+	buildNumber := "1"
+	oldHomeDir, newHomeDir := prepareHomeDir(t)
+	defer os.Setenv(coreutils.HomeDir, oldHomeDir)
+	defer os.RemoveAll(newHomeDir)
+	wd, err := os.Getwd()
+	assert.NoError(t, err)
+	prepareGoProject("", t, true)
+	// Build and publish a go project
+	runGo("", tests.GoBuildName, buildNumber, t, "go", "build", "--build-name="+tests.GoBuildName, "--build-number="+buildNumber)
```
Comment (eyalbe4) [inline go_test.go:164]: We need to check if this "go build" is indeed required, and if so, explain why in a comment.

File: go_test.go [hunk 2]
```diff
@@ -143,12 +145,39 @@ func TestGoWithGlobalConfig(t *testing.T) {
...
+
+	// Act
+	// Go get one of the known pushed dependencies
+	err = tests.NewJfrogCli(execMain, "jfrog rt", "").Exec("go", "get", "rsc.io/quote@v1.5.2", "--build-name="+tests.GoBuildName, "--build-number="+buildNumber)
+	assert.NoError(t, err)
+	artifactoryCli.Exec("bp", tests.GoBuildName, buildNumber)
+	buildInfo, _ := inttestutils.GetBuildInfo(artifactoryDetails.Url, tests.GoBuildName, buildNumber, t, artHttpDetails)
+
+	//Assert
```
Comment (eyalbe4) [inline go_test.go:173]: Let's remove the "//Assert" comment, because currently it is currently not part of our testing code conventions.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: go_test.go [hunk 3]
```diff
@@ -143,12 +145,39 @@ func TestGoWithGlobalConfig(t *testing.T) {
...
+	// Init test and preper Global config
+	initGoTest(t)
+	buildNumber := "1"
+	oldHomeDir, newHomeDir := prepareHomeDir(t)
+	defer os.Setenv(coreutils.HomeDir, oldHomeDir)
+	defer os.RemoveAll(newHomeDir)
+	wd, err := os.Getwd()
+	assert.NoError(t, err)
+	prepareGoProject("", t, true)
+	// Build and publish a go project
```
[the remark this answers is not in the document]
Comment (talarian1) [reply go_test.go:163]: runGo execute "bp" after executing the given command. So it is building and publishing a go project...

File: go_test.go (hunk 1 above)
Comment (talarian1) [reply go_test.go:164]: It is needed. I added a more detailed comment.

**Pull request merged**